src/groupeddataframe/grouping.jl

```diff
@@ -122,50 +119,111 @@ _names(gd::GroupedDataFrame) = _names(gd.parent)
 # Wrapping automatically adds column names when the value returned
 # by the user-provided function lacks them
 wrap(x::Union{AbstractDataFrame, NamedTuple, DataFrameRow}) = x
-wrap(x::AbstractMatrix) = convert(DataFrame, x)
-wrap(x::AbstractVector) = DataFrame(x1 = x)
-wrap(x::Any) = (x1 = x,)
+wrap(x::AbstractMatrix) =
+    NamedTuple{Tuple(gennames(size(x, 2)))}(Tuple(x[:, i] for i in 1:size(x, 2)))
```

Comment:
do we want to create a copy or a view here? I am referring to `x[:, i]`.

Reply:
Any opinion on this? I am OK to leave it as is, but want this to be an explicit decision 😄.

Reply:
Indeed views are more appropriate since we're going to make copies anyway when allocating the final column.